### PR TITLE
feat: connection pooling and idle timeout

### PR DESCRIPTION
**Pull request by peterdeme**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -32,9 +32,6 @@ Metrics/PerceivedComplexity:
 Naming/AccessorMethodName:
   Enabled: false
 
-Security/Open:
```
Comment (ferhatelmas) [inline .rubocop.yml:35]: reason, http_persistent?

Comment (peterdeme) [reply .rubocop.yml:35]: Not related.

- We don't use open anywhere
- I don't like having security-related issues ignored

Comment (ferhatelmas) [reply .rubocop.yml:35]: > I don't like having security-related issues ignored

this is the mindset 💯

Comment (ferhatelmas) [reply .rubocop.yml:35]: > We don't use open anywhere

why we ignored it before is a question though

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**